### PR TITLE
feat: add elastic training example

### PR DESCRIPTION
**Pull request by shjwudp**

### CODE REVIEW

File: elastic_training/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Use the following script to start elastic training with x node * 8 gpus:
```
Comment (NOBLES5E) [inline elastic_training/README.md]: what is x

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: elastic_training/README.md (hunk 1 above)
Comment (shjwudp) [reply elastic_training/README.md]: two, fixed.

File: benchmark/synthetic_benchmark.py [hunk 2]
```diff
@@ -20,10 +20,8 @@
     formatter_class=argparse.ArgumentDefaultsHelpFormatter,
 )
 
-parser.add_argument("--model", type=str, default="resnet50",
-                    help="model to benchmark")
-parser.add_argument("--batch-size", type=int,
-                    default=32, help="input batch size")
+parser.add_argument("--model", type=str, default="resnet50", help="model to benchmark")
```
Comment (NOBLES5E) [inline benchmark/synthetic_benchmark.py:23]: not related to this pr, make it another one

File: elastic_training/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Use the following script to start elastic training with 2 node * 8 gpus:
+
+Node 1: *(IP: 192.168.1.1, and has a free port: 1234)*
+
+```bash
+python3 -m bagua.distributed.run \
+        --nnodes=1:4 \
```
Comment (NOBLES5E) [inline elastic_training/README.md:7]: 2 nodes, but nnodes=1:4, is this expected?

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: benchmark/synthetic_benchmark.py (hunk 2 above)
Comment (shjwudp) [reply benchmark/synthetic_benchmark.py:23]: Just black format the code. It is necessary and hard to recover..

File: elastic_training/README.md (hunk 3 above)
Comment (shjwudp) [reply elastic_training/README.md:7]: Yes it is. document changed.

**Pull request merged**